api/src/main/java/com/epam/esm/controller/GiftController.java

```diff
@@ -0,0 +1,94 @@
+package com.epam.esm.controller;
+
+import com.epam.esm.model.CustomSearchRequest;
+import com.epam.esm.model.dto.GiftCertificateDto;
+import com.epam.esm.service.GiftService;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/v1/gifts")
+public class GiftController {
+    private final GiftService giftService;
+    private final CustomSearchRequest defaultCustomSearchRequest;
+
+
+    @Autowired
+    public GiftController(GiftService giftService) {
+        this.giftService = giftService;
+        defaultCustomSearchRequest = CustomSearchRequest.builder().build();
```

Comment:
Actually, it will be similar to you approach. However, usually such field are made as a constant variable